core/new-gui/src/app/home/component/login/local-login/local-login.component.ts

```diff
@@ -62,19 +65,22 @@ export class LocalLoginComponent {
 
     this.userService
       .login(username, password)
-      .pipe(untilDestroyed(this))
-      .subscribe({
-        next: () => {
-          window.location.href = this.route.snapshot.queryParams["returnUrl"] || "/dashboard/workflow";
-        },
-        error: (err: unknown) => {
+      .pipe(
+        catchError((err: unknown) => {
           if (err instanceof HttpErrorResponse) {
             this.notificationService.error(err.error.message, {
               nzDuration: 0,
             });
           }
-        },
-      });
+          return throwError(() => err);
+        }),
+        untilDestroyed(this)
+      )
+      .subscribe(
+        Zone.current.wrap(() => {
```

Comment:
I remember we had a comment here explaining the reason to use Zone as a wrapper. Can we add it back for future reference?

Reply:
Added.